### PR TITLE
test(unhappy): mask `sender` when comparing timeout_qcs

### PR DESCRIPTION
**Pull request by youngjoon-lee**
At the end of the unhappy test, we sometimes check if all nodes have the same timeout_qc. But, this check can be failed if timeout_qcs are from different senders, even though all other fields are the same.

In this check, we want to consider those timeout_qcs as the same timeout_qc.

- 66f80fc25da037cf74d5006665a7975030336b1d: Refactoring
- fcd48aa: Core changes

### CODE REVIEW

**Review by zeegomo — APPROVED** [review]
I'm actually wondering if multiple members of the root committee producing timeout_qcs could cause synchronization problems where different timeout_qcs could be built from different timeouts, thus possibly influencing the way the next leader is chosen. We currently are not sensitive to that but it may change in the future.
Anyway, this is change is ok

**Comment by danielSanchezQ** [comment]
> I'm actually wondering if multiple members of the root committee producing timeout_qcs could cause synchronization problems where different timeout_qcs could be built from different timeouts, thus possibly influencing the way the next leader is chosen. We currently are not sensitive to that but it may change in the future. Anyway, this is change is ok.

Uhm, shouldn't all generated timeout_qcs be the same? The only difference would be the issuer right?
As we use the random beacon to generate them shouldn't it be deterministic? if one of them generates a wrong qc it would not be valid.
I may be completely wrong here.

**Comment by zeegomo** [comment]
> Uhm, shouldn't all generated timeout_qcs be the same?

They could be generated using as proof a different subset of the root committee timeouts. The rest should be equal, but it's almost guaranteed that the subset will be different because the latencies will be different (for sure all nodes will include their own timeout)

**Comment by danielSanchezQ** [comment]
> > Uhm, shouldn't all generated timeout_qcs be the same?
> 
> They could be generated using as proof a different subset of the root committee timeouts. The rest should be equal, but it's almost guaranteed that the subset will be different because the latencies will be different (for sure all nodes will include their own timeout)

Oooh, ok, I get it know. I think it should be ok anyway if we exclude issuers from comparisons/checks.

**Comment by youngjoon-lee** [comment]
Aha, I get it. Thank you for the explanation. I also think it’s ok if we exclude senders from comparisons.

File: tests/src/tests/unhappy.rs [hunk 1]
```diff
@@ -46,30 +48,67 @@ async fn ten_nodes_one_down() {
...
+    }
+}
 
-        let timeout_qcs = infos
-            .iter()
-            .map(|i| i.last_view_timeout_qc.clone())
-            .collect::<HashSet<_>>();
-        assert_eq!(timeout_qcs.len(), 1);
+// Check if all nodes have the same block at the specific view.
+fn assert_block_consensus(consensus_infos: Iter<CarnotInfo>, view: View) -> Option<Block> {
```
[the remark this answers is not in the document]
Comment (youngjoon-lee) [reply tests/src/tests/unhappy.rs]: Thank you. I just changed it to use `impl IntoIterator<Item = ...>`: 82f8bc1. But, I had to clone the vector once for that. The previous solution may use less clones, but I think it's not a big deal in testing. What do you think about this?

[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply tests/src/tests/unhappy.rs]: A slice reference would do as well. Also you can use `cloned` in the iterator. But it is not really important.

[the remark this answers is not in the document]
Comment (youngjoon-lee) [reply tests/src/tests/unhappy.rs]: Thank you! 982c371. I actually tried to use references  at the beginning but I did something wrong :) Now, it works. Thank you for your help.

**Review by danielSanchezQ — APPROVED** [review]

**Pull request merged**